server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DeleteTemplateAction.java

```diff
@@ -76,21 +78,50 @@ private void doHandle(DeleteTemplateWsRequest request) {
         request.getTemplateId(), request.getOrganization(), request.getTemplateName()));
       checkGlobalAdmin(userSession, template.getOrganizationUuid());
 
-      checkTemplateUuidIsNotDefault(template.getUuid());
+      DefaultTemplates defaultTemplates = retrieveDefaultTemplates(dbSession, template);
+
+      checkTemplateUuidIsNotDefault(template, defaultTemplates);
       dbClient.permissionTemplateDao().deleteById(dbSession, template.getId());
+      updateViewDefaultTemplateWhenGovernanceIsNotInstalled(dbSession, template, defaultTemplates);
+
       dbSession.commit();
     }
   }
 
+  /**
+   * The default template for view can be removed when Governance is not installed. To avoid keeping a reference
+   * to a non existing template, we update the default templates.
+   */
+  private void updateViewDefaultTemplateWhenGovernanceIsNotInstalled(DbSession dbSession, PermissionTemplateDto template, DefaultTemplates defaultTemplates) {
```

Comment:
BTW, keeping the reference would mean, that once you reinstall governance, SQ will attempt to apply the non existing permission template when created views and will fail.

Lastly, coming back to the use case you reference, if governance is disabled just to fix the license, why would one delete the permission template registered as the default for views?